scripts/zones/Rolanberry_Fields/npcs/Mayuyu.lua

```diff
@@ -0,0 +1,66 @@
+-----------------------------------
+-- Area: Rolanberry Fields
+--  NPC: Mayuyu
+-- Legion NPC
+-- @pos 240 24.399 468
+-----------------------------------
+package.loaded["scripts/zones/Rolanberry_Fields/TextIDs"] = nil;
+-----------------------------------
+
+require("scripts/globals/settings");
+require("scripts/globals/keyitems");
+require("scripts/globals/titles");
+require("scripts/zones/Rolanberry_Fields/TextIDs");
+
+-----------------------------------
+-- onTrade Action
+-----------------------------------
+
+function onTrade(player,npc,trade)
+end;
+
+-----------------------------------
+-- onTrigger Action
+-----------------------------------
+function onTrigger(player,npc)
+    local LEGENDARY = 0;
+    local CAPACITY = 0;
+
+    if (player:hasTitle(LEGENDARY_LEGIONNAIRE)) then
+        LEGENDARY = 1;
+    end
+
+    if (player:hasKeyItem(LEGION_TOME_PAGE_MAXIMUS)) then
+        CAPACITY = CAPACITY+2; -- Enable 36 combatant option
+    end
+    if (player:hasKeyItem(LEGION_TOME_PAGE_MINIMUS)) then
+        CAPACITY = CAPACITY+1; -- Enable 18 combatant option
+    end
+
+    if (player:getMainLvl() < 75) then
+        player:showText(npc, 12131);
+    else
+        player:startEvent(8008, 0, LEGENDARY, CAPACITY);
+    end
+end;
+
+-----------------------------------
+-- onEventUpdate
+-----------------------------------
+
+function onEventUpdate(player,csid,option)
+    -- print("CSID:", csid);
+    -- print("RESULT:", option);
+end;
+
+-----------------------------------
+-- onEventFinish
+-----------------------------------
+
+function onEventFinish(player,csid,option)
+    -- print("CSID:", csid);
+    -- print("RESULT:", option);
+
+    -- Event needs work, also the Legion Pass item is "tagged" via fields not yet implemented in core.

```

Comment:
it's all part of m_extra of CItem, you should be able to find everything you need for that at https://github.com/Windower/Lua/blob/4.1-dev/addons/libs/extdata.lua